pkg/controller/keyvault/placeholder_pod.go

```diff
@@ -178,6 +223,12 @@ func (p *PlaceholderPodController) buildDeployment(ctx context.Context, dep *app
 	if old != nil { // we need to ensure that immutable fields are not changed
 		labels = old.Spec.Selector.MatchLabels
 	}
+	var ingAnnotation string
+	if _, ok := obj.(*v1alpha1.NginxIngressController); ok {
+		ingAnnotation = "kubernetes.azure.com/nginx-ingress-controller-owner"
+	} else {
+		ingAnnotation = "kubernetes.azure.com/ingress-owner"
+	}
```

Comment:
switch with a default case of returning error. what if obj isn't one of these two types?